src/test/java/spoon/test/compilationunit/TestCompilationUnit.java

```diff
@@ -127,4 +133,24 @@ public void testAddDeclaredTypeInCU() throws IOException {
         assertTrue(fullContent.contains("class Baz"));
         assertTrue(fullContent.contains("class Bla"));
     }
+
+    @Test
+    public void testNewlyCreatedCUWouldGetAPartialPosition() throws IOException {
+        // contract: when a type is created, a CU can be created and added as partial position
+        final Launcher launcher = new Launcher();
+        assertTrue(launcher.getFactory().CompilationUnit().getMap().isEmpty());
+
+        CtClass myNewClass = launcher.getFactory().createClass("my.new.MyClass");
+        assertEquals(SourcePosition.NOPOSITION, myNewClass.getPosition());
+
+        CompilationUnit cu = launcher.getFactory().CompilationUnit().getOrCreate(myNewClass);
+
+        assertNotNull(cu);
+        SourcePosition sourcePosition = myNewClass.getPosition();
+        assertTrue(sourcePosition instanceof PartialSourcePositionImpl);
+        assertEquals(cu, sourcePosition.getCompilationUnit());
```

Comment:
replace assertEquals -> assertSame

Reply:
I'm not quite sure about changing this one: in the current implementation it's true, but I don't think it should be a contract. We can imagine implementation in which it can return a clone of CompilationUnit for example.

Reply:
> return a clone

sourcePosition is an unmodifiable element. There are no setters on it. It should always return the same values. I think it is the contract.

Reply:
> sourcePosition is an unmodifiable element.

Yes but you can change the SourcePosition object itself... But ok I'm not sure on this one: I'll change it if you think it should be a contract.